src/analyzer/vamp/vamppluginadapter.cpp

```diff
@@ -137,37 +134,133 @@ void initPluginPaths() {
 #endif
 }
 
-void initPluginLoader() {
-    initPluginPaths();
-    s_pPluginLoader = Vamp::HostExt::PluginLoader::getInstance();
+std::mutex s_mutex;
```

Comment:
It looks like there are some reasons to prefer QMutex over std::mutex:
https://stackoverflow.com/questions/29170495/performance-of-stdmutex-and-qmutex-in-mingw-64-posix-thread-version

Reply:
http://man7.org/linux/man-pages/man7/pthreads.7.html
"...In NPTL, thread synchronization primitives (mutexes, thread joining, and so on) are implemented using the Linux futex(2) system call."

I have more trust in the standard implementation that comes with a broader background and audience. I suppose QMutex will once become a simple wrapper around std::mutex. There is no more reason to maintain a custom solution. That's legacy and history.

Reply:
Do QObject::thread, QObject::moveToThread work with std::thread? How do you get a QThread* handle to a std::thread to pass to those?

What about queued connections where the received "lives" in a std::thread?